src/charm.py

```diff
@@ -110,59 +71,42 @@ def _on_github_actions_exporter_pebble_ready(self, event: WorkloadEvent):
         container.add_layer(container.name, self._pebble_layer, combine=True)
         container.replan()
         self.unit.status = ActiveStatus()
-        self.unit.set_workload_version(self._get_exporter_version())
+        self.unit.set_workload_version(self._github_actions_exporter.get_exporter_version())
 
     def _is_configuration_valid(self) -> bool:
```

Comment:
Should this be checked as a part of creating the state? Weii has a good example in the 12-factor flask charm repo for it

Reply:
Moved to Github exporter file.